### DISCUSSION TITLE
Can I convert Chimap.nii.gz into DICOM format?

### DISCUSSION THREAD

**Discussion by kaiyuliu123** [kschan0214/sepia#78] [General]
Hi, I want to know if it is possible to convert QSM outcome(Chimap.nii.gz) into DICOM file after I've gone through the Sepia pepline. I tried to load my Chimap in 3Dslier and export it to dicom but failed. The dicom directory generated was filled with blank images. Could someone offer some advice regarding this issue since dcm files are more practical in clinics and are sometimes the only format allowed in some steretaxic surgery planning systems such as Brainlab.
[Sepia_Chimap.nii.gz](https://github.com/kschan0214/sepia/files/13915076/Sepia_Chimap.nii.gz)

**Comment by kschan0214** [comment]
Hi, Sorry for the late reply. I haven't tried any NIFTI to DICOM tools before but I guess this might be related to the image data type in the NIFTI file. By default, the data type of the SEPIA's output is single precision (32-bit floating point), which might not be directly supported in DICOM format (usually 16-bit integer). 

One option I can think of is to convert the data from float32 to int16 type using the following code:

```
%% Convert float32 nifti to int16 nifti
sepia_addpath;

input_filename  = 'Sepia_Chimap.nii.gz';
output_filename = 'Sepia_Chimap_datatype-int16.nii.gz';

nii = load_untouch_nii(input_filename);
img_nifti = nii.img;

% get current image limit and new image limit
maxValue_nifti = max(img_nifti(:));
minValue_nifti = min(img_nifti(:));
maxValue_dicom = single(intmax("int16"));
minValue_dicom = single(intmin("int16"));

% compute rescale slope and intercept
rescale_slope       = (maxValue_nifti-minValue_nifti) / (maxValue_dicom - minValue_dicom);
rescale_intercept   = -minValue_dicom * rescale_slope + minValue_nifti;

% convert data from float 32 to int16
img_dicom = int16((img_nifti - minValue_nifti) / (maxValue_nifti-minValue_nifti) * (maxValue_dicom - minValue_dicom) + minValue_dicom);

% update nifti header
nii.img = img_dicom;
nii.hdr.dime.bitpix     = 16;
nii.hdr.dime.datatype   = 4;
nii.hdr.dime.scl_inter = rescale_intercept;
nii.hdr.dime.scl_slope = rescale_slope;
nii.hdr.dime.glmax = maxValue_dicom;
nii.hdr.dime.glmin = minValue_dicom;

% export new nifti file
save_untouch_nii(nii, output_filename)
```
[convert_float32_2_int16.m.zip](https://github.com/kschan0214/sepia/files/14067760/convert_float32_2_int16.m.zip)

then convert the data to DICOM again using the tool. Note that there could be a precision change after the conversion due to the change in data type (32-bit -> 16-bit).

Let me know if it works!
